cart/domain/cart/cart.go

```diff
@@ -601,6 +628,30 @@ func (t Taxes) TotalAmount() domain.Price {
 	return result
 }
 
+// GetAllTaxes – returns all Taxes as slice
+func (t Taxes) GetAllTaxes() []Tax {
+	taxes := make([]Tax, 0, len(t))
+	for _, tax := range t {
+		taxes = append(taxes, tax)
+	}
+
+	if len(taxes) > 0 {
+		return taxes
+	}
+
+	return nil
+}
+
+// GetByType - returns tax by given type
+func (t Taxes) GetByType(taxType string) Tax {
```

Comment:
Because Tax can not exist for a type imho makes sense to return pointer and nil in case its not found